src/middlewared/middlewared/plugins/auth.py

```diff
@@ -325,9 +339,128 @@ def token(self, app, token):
         return True
 
 
+class TwoFactorAuthService(ConfigService):
+
+    class Config:
+        datastore = 'system.twofactorauthentication'
+        datastore_extend = 'auth.twofactor.two_factor_extend'
+        namespace = 'auth.twofactor'
+
+    @private
+    async def two_factor_extend(self, data):
+        data['secret'] = await self.middleware.call('pwenc.decrypt', data['secret'])
+
+        for srv in ['ssh']:
+            if not data['services'].get(srv):
```

Comment:
Lifehack: `data['services'].setdefault(srv, False)` (not asking you to change it this time, just for the future)